xbox/webapi/authentication/manager.py

```diff
@@ -2,641 +2,147 @@
-                self.title_token = t
-            elif isinstance(t, XSTSToken) and should_replace(self.xsts_token):
-                self.xsts_token = t
-
-        file_userinfo = json_file.get('userinfo')
-        if not self.userinfo and file_userinfo:
-            self.userinfo = XboxLiveUserInfo.from_dict(file_userinfo)
-
-    def dump(self, filepath):
-        """
-        Save tokens and userinfo as json file
-
-        Args:
-            ts (object): Instance of :class:`Tokenstore`
-
-        Returns:
-            None
-        """
-        json_file = dict(tokens=list(), userinfo=None)
-
-        tokens = [self.access_token, self.refresh_token, self.user_token,
-                  self.xsts_token, self.title_token, self.device_token]
-        for token in tokens:
-            if not token:
-                continue
-            json_file['tokens'].append(token.to_dict())
-
-        if self.userinfo:
-            json_file['userinfo'] = self.userinfo.to_dict()
-
-        with open(filepath, 'w') as f:
-            json.dump(json_file, f, indent=2)
-
-    @staticmethod
-    def generate_authorization_url(response_type='token', client_id='0000000048093EE3',
-                                   scope='service::user.auth.xboxlive.com::MBI_SSL',
-                                   redirect_uri='https://login.live.com/oauth20_desktop.srf',
-                                   state=None):
-        """
-        Generate Windows Live Authorization URL.
-
-        Either standard values can be used, to authentication to XBL and get a XToken, or use custom
-        parameters to authenticate with a specific service (Halo, Forza ... you name it)
-
-        Args:
-            response_type (str): Required authorization response, either 'code' or 'token'
-            client_id (str): Client ID of service to authenticate with
-            scope (str): Authorization scope
-            redirect_uri (str): URL to redirect to when authentication succeeds
-            state (str): Optional, OAuth2 state url
-
-        Returns:
-            str: Assembled URL, including query parameters
-        """
-
-        base_url = 'https://login.live.com/oauth20_authorize.srf'
-        params = {
-            'client_id': client_id,
-            'redirect_uri': redirect_uri,
-            'response_type': response_type,
-            'display': 'touch',
-            'scope': scope,
-            'locale': 'en',
+from typing import List, Optional
+
+import aiohttp
+from yarl import URL
+
+from xbox.webapi.authentication.models import (
+    OAuth2TokenResponse,
+    XAUResponse,
+    XSTSResponse,
+)
+
+log = logging.getLogger("authentication")
+
+DEFAULT_SCOPES = ["Xboxlive.signin", "Xboxlive.offline_access"]
+
+
+class AuthenticationManager:
+    def __init__(
+        self,
+        client_session: aiohttp.ClientSession,
+        client_id: str,
+        client_secret: str,
+        redirect_uri: str,
+        scopes: List[str] = DEFAULT_SCOPES,
```

Comment:
Consider setting the default scopes argument to `None` and replacing that to `DEFAULT_SCOPES` in the `__init__` method instead. Using a mutable object as default argument is bug prone.